File: InAppChat/src/main/java/io/inappchat/inappchat/UserConfiguration.java
package io.inappchat.inappchat;

import android.content.Context;
import androidx.annotation.NonNull;

/** Created by Sagar on 02/05/22. */
public class UserConfiguration {

  private final String name;
  private final String status;
  private final String appUserId;
  private final String loginType;
  private final String profilePic;
  private final String profilePicThumb;
  private final String profileStatus;
  private final String appState;
  private final String userId;
  private final String accessToken;
  private final String refreshToken;
  private final String fcmToken;
  private final String payload;
  private final Context context;

  private UserConfiguration(UserConfiguration.Builder builder) {
    this.name = builder.name;
    this.status = builder.status;
    this.appUserId = builder.appUserId;
    this.loginType = builder.loginType;
    this.profilePic = builder.profilePic;
    this.profilePicThumb = builder.profilePicThumb;
    this.profileStatus = builder.profileStatus;
    this.appState = builder.appState;
    this.userId = builder.userId;
    this.accessToken = builder.accessToken;
    this.refreshToken = builder.refreshToken;
    this.fcmToken = builder.fcmToken;
    this.payload = builder.payload;
    this.context = builder.context;
  }

  public String getName() {
    return name;
  }

  public String getStatus() {
    return status;
  }

  public String getAppUserId() {
    return appUserId;
  }

  public String getLoginType() {
    return loginType;
  }

  public String getProfilePic() {
    return profilePic;
  }

  public String getProfilePicThumb() {
    return profilePicThumb;
  }

  public String getProfileStatus() {
    return profileStatus;
  }

  public String getAppState() {
    return appState;
  }

  public String getUserId() {
    return userId;
  }

  public String getAccessToken() {
    return accessToken;
  }

  public String getRefreshToken() {
    return refreshToken;
  }

  public String getFcmToken() {
    return fcmToken;
  }

  public String getPayload() {
    return payload;
  }

  public Context getContext() {
    return context;
  }

  public static class Builder {
    private String name;
    private String status;
    private String appUserId;
    private String loginType;
    private String profilePic;
    private String profilePicThumb;
    private String profileStatus;
    private String appState;
    private String userId;
    private String accessToken;
    private String refreshToken;
    private String fcmToken;
    private String payload;
    private Context context;

    public Builder() {}

    public UserConfiguration.Builder name(@NonNull String name) {
      this.name = name;
      return this;
    }

    public UserConfiguration.Builder status(@NonNull String status) {
      this.status = status;
      return this;
    }

    public UserConfiguration.Builder appUserId(@NonNull String appUserId) {
      this.appUserId = appUserId;
      return this;
    }

    public UserConfiguration.Builder loginType(@NonNull String loginType) {
      this.loginType = loginType;
      return this;
    }

    public UserConfiguration.Builder profilePic(@NonNull String profilePic) {
      this.profilePic = profilePic;
      return this;
    }

    public UserConfiguration.Builder profilePicThumb(@NonNull String profilePicThumb) {
      this.profilePicThumb = profilePicThumb;
      return this;
    }

    public UserConfiguration.Builder profileStatus(@NonNull String profileStatus) {
      this.profileStatus = profileStatus;
      return this;
    }

    public UserConfiguration.Builder appState(@NonNull String appState) {
      this.appState = appState;
      return this;
    }

    public UserConfiguration.Builder userId(@NonNull String userId) {
      this.userId = userId;
      return this;
    }
    public UserConfiguration.Builder accessToken(@NonNull String accessToken) {
      this.accessToken = accessToken;
      return this;
    }

    public UserConfiguration.Builder refreshToken(@NonNull String refreshToken) {
      this.refreshToken = refreshToken;
      return this;
    }

    public UserConfiguration.Builder fcmToken(@NonNull String fcmToken) {
      this.fcmToken = fcmToken;
      return this;
    }

    public UserConfiguration.Builder payload(@NonNull String payload) {
      this.payload = payload;
      return this;
    }

    public UserConfiguration.Builder context(@NonNull Context context) {
      this.context = context;
      return this;
    }

    public UserConfiguration build() {
      return new UserConfiguration(this);
    }
  }
}

File: InAppChat/src/main/java/io/inappchat/inappchat/UserConfiguration.kt
package io.inappchat.inappchat

import android.content.Context

class UserConfiguration constructor(builder: Builder) {
    val name: String?
    val status: String?
    val appUserId: String?
    val loginType: String?
    val profilePic: String?
    val profilePicThumb: String?
    val profileStatus: String?
    val appState: String?
    val userId: String?
    val accessToken: String?
    val refreshToken: String?
    val fcmToken: String?
    val payload: String?
    val context: Context?

    init {
        name = builder.name
        status = builder.status
        appUserId = builder.appUserId
        loginType = builder.loginType
        profilePic = builder.profilePic
        profilePicThumb = builder.profilePicThumb
        profileStatus = builder.profileStatus
        appState = builder.appState
        userId = builder.userId
        accessToken = builder.accessToken
        refreshToken = builder.refreshToken
        fcmToken = builder.fcmToken
        payload = builder.payload
        context = builder.context
    }

    class Builder {
        var name: String? = null
        var status: String? = null
        var appUserId: String? = null
        var loginType: String? = null
        var profilePic: String? = null
        var profilePicThumb: String? = null
        var profileStatus: String? = null
        var appState: String? = null
        var userId: String? = null
        var accessToken: String? = null
        var refreshToken: String? = null
        var fcmToken: String? = null
        var payload: String? = null
        var context: Context? = null
        fun name(name: String): Builder {
            this.name = name
            return this
        }

        fun status(status: String): Builder {
            this.status = status
            return this
        }

        fun appUserId(appUserId: String): Builder {
            this.appUserId = appUserId
            return this
        }

        fun loginType(loginType: String): Builder {
            this.loginType = loginType
            return this
        }

        fun profilePic(profilePic: String): Builder {
            this.profilePic = profilePic
            return this
        }

        fun profilePicThumb(profilePicThumb: String): Builder {
            this.profilePicThumb = profilePicThumb
            return this
        }

        fun profileStatus(profileStatus: String): Builder {
            this.profileStatus = profileStatus
            return this
        }

        fun appState(appState: String): Builder {
            this.appState = appState
            return this
        }

        fun userId(userId: String): Builder {
            this.userId = userId
            return this
        }

        fun accessToken(accessToken: String): Builder {
            this.accessToken = accessToken
            return this
        }

        fun refreshToken(refreshToken: String): Builder {
            this.refreshToken = refreshToken
            return this
        }

        fun fcmToken(fcmToken: String): Builder {
            this.fcmToken = fcmToken
            return this
        }

        fun payload(payload: String): Builder {
            this.payload = payload
            return this
        }

        fun context(context: Context): Builder {
            this.context = context
            return this
        }

        fun build(): UserConfiguration {
            return UserConfiguration(this)
        }
    }
}